config/configmap.go

```diff
@@ -40,31 +35,6 @@ func NewMap() *Map {
 	return &Map{k: koanf.New(KeyDelimiter)}
 }
 
-// NewMapFromFile creates a new config.Map by reading the given file.
-func NewMapFromFile(fileName string) (*Map, error) {
```

Comment:
Is it worth keeping NewMapFrom[File|Buffer] and making them private to simplify the code? Maybe not as they're used across many different packages which wouldn't be able to use them anyways.

Reply:
That is the think, they are not used too much, so I think we can use the provider, but only in 2 tests we cannot.